src/testdrive/src/action/kafka/ingest.rs

```diff
@@ -174,45 +191,79 @@ impl Action for IngestAction {
 
         let futs = FuturesUnordered::new();
         for row in &self.rows {
-            let mut buf = Vec::new();
+            let mut valbuf = Vec::new();
+            let mut keybuf = Vec::new();
             match &encoder {
-                Encoder::Avro { schema, schema_id } => {
+                Encoder::Avro {
+                    key_encode,
+                    value_schema,
+                    value_schema_id,
+                } => {
+                    let (keyrow, valrow) = if key_encode.is_some() {
+                        let tokens: Vec<_> = row.split('␉').collect();
```

Comment:
I would very much prefer not to use ␉here. The fact that we need to use the equivalent NL character in SLT is an unfortunate hack required for backwards compatibility with the large corpus of existing SLT tests. Since we control all the testdrive tests, we can do something more elegant here.

Instead I propose we use a stream deserializer: https://docs.serde.rs/serde_json/struct.StreamDeserializer.html. Then the key and value data can be separated by any whitespace (or nothing at all, though we can frown on that style).